lib/spack/spack/test/monitor.py

```diff
@@ -48,7 +48,7 @@ def mock_monitor_request(monkeypatch):
     def mock_do_request(self, endpoint, *args, **kwargs):
 
         build = {"build_id": 1,
-                 "spec_full_hash": "bpfvysmqndtmods4rmy6d6cfquwblngp",
+                 "spec_hash": "bpfvysmqndtmods4rmy6d6cfquwblngp",
```

Comment:
yep. agree. There are two kinds of changes in this PR - to the mock of spack monitor, and to the actual request endpoints.  Changing this specific `mock_do_request` function, for example, doesn't make sense because it's mocking spack monitor, which will not return spec_hash but rather `spec_full_hash`. It might still work within the tests here but it's no longer accurate for what spack monitor will actually do. 

For other requests here the important thing is that spack monitor is going to look for `full_hash` as part of the build environment [here](https://github.com/spack/spack-monitor/blob/5c2f46e7be722fc61ba2c6f838d818437cc97327/spackmon/apps/api/views/builds.py#L46) so changing it to "hash" across the functions is going to break things. Specifically this is going to break things:

- https://github.com/spack/spack/blob/da13102483421b8fdd6f946ba7fa4aed264ab256/lib/spack/spack/monitor.py#L440-L441 because spack monitor needs `full_hash` and doesn't care for hash or runtime hash. This is going to result in missing data and [fail to accurately show statuses](https://github.com/spack/spack-monitor/blob/95a6e62503bc3f4d0a591ca84fbe37c3f90b256e/spackmon/apps/main/models.py#L702-L712).
- https://github.com/spack/spack/blob/da13102483421b8fdd6f946ba7fa4aed264ab256/lib/spack/spack/monitor.py#L479 because it's going to look for full hash.

If the new spec doesn't have full hash it might still generate an index error, [here](https://github.com/spack/spack-monitor/blob/5c2f46e7be722fc61ba2c6f838d818437cc97327/spackmon/apps/main/tasks.py#L201-L203). I would go as far to say that (for now) the spack monitor endpoints should add it.

My suggestion for this PR is to not make changes to spack monitor, but rather ensure that the same data is had before (the full hash) is still supplied, and no big deal if it happens to be a different (dag) hash. And generally you can't change the server mock unless you change the server first.